_posts/2020-11-08-basic-coding-applications-for-commercial-strategy.md

```diff
@@ -0,0 +1,83 @@
+
+
+# Data Cleaning
+Before we dive into each code example, it is crucial to clean your raw data first. The most important items to check are that the column titles are in line with what your code references as well as to check that the data points are consistently and correctly formatted.
+1. **Check raw data column titles:** The raw data column titles must match the code exactly including capitalization. With code that references `revenue`, a column title of `Revenue` will not work. If your general code references `territory`, but your raw data uses `states` as the territory descriptor, the column title for the states must be changed to `territory`.
+2. **Check data format consistency:** If a data point doesn’t exist, it needs to say `NA`. Otherwise, the data should follow the same template throughout the data set and that aligns with the code. For example, the state of California should not be represented as `California`, `california`, `CA`, and `ca`. It doesn't matter which one is used as long as it is the same format used for all raw data points and the code input.
+
+# Directory Set-Up
+Create your project folder with your R project file saved into this folder. Also include the following folders: `data`, `code`, `output`. The `data` folder should be the location where all of your raw data files are kept. The `code` folder should contain your scripts. The `output` folder is where your output files should be saved to and should also include a subfolder titled `figures`.
+
+# Applications
+## Territory Analysis
+It is essential to analyze revenue by territory for many different reasons. Each territory can have a slightly different client base that has different needs that should be understood in order to create a successful commercial strategy specific to that region. Additionally, if different territories are covered by different reps, a clear understanding of territory performance is essential for quota development and commission payout. Along the same lines, territory splits can be determined should certain territories show growth that warrants multiple account managers.
+#### Starting Material:
+A revenue report that has the following columns should be saved in your `data` folder in your directory -
+* `revenue` - revenue dollar amount
+* `date` - invoice date, shipping date, start date, purchase date, or whatever date is most relevant for your purposes
+* `territory` - sales representative (i.e. Debbie Vo), state (i.e. WA), territory name (i.e. North West)
+
+#### Code to Separate Data by Territory:
+We start off by creating a general function to subset any territory out of the full data set. This code should not have to change when different files and territories are indicated. This should also create an output CSV file that is named based on the defined territory.
+```ruby
```

Comment:
It would be useful if you could describe Markdown processing utilizing ruby and what sort of programming language it is.